sdk/typescript/src/cryptography/signature.ts

```diff
@@ -59,8 +62,16 @@ export function fromSerializedSignature(
   const signatureScheme =
     SIGNATURE_FLAG_TO_SCHEME[bytes[0] as keyof typeof SIGNATURE_FLAG_TO_SCHEME];
 
-  const PublicKey =
-    signatureScheme === 'ED25519' ? Ed25519PublicKey : Secp256k1PublicKey;
+  const PublicKey = (function () {
```

Comment:
Instead of doing htis, can we make an object that maps from signature scheme to public key? Basically just:

```typescript
const SCHEME_TO_KEY = {
  ED25519: Ed25519PublicKey,
  // ... the rest
}
```

Then you can just look up in this:


```typescript
const PublicKey = SCHEME_TO_KEY[signatureScheme];
```